(test) O3-3863: E2E tests for user onboarding tutorials

harshthakkr opened the pull request:
## Requirements
- [x] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [x]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [x] My work includes tests or is validated by existing tests.

## Summary
This PR adds E2E tests for the following tutorials:
1. Basic overview
2. Demo tutorial
3. Finding a patient
4. Patient chart
5. Recording vitals
6. Registering a patient
7. Starting a patient visit

## Screenshots

## Related Issue
[O3-3863](https://openmrs.atlassian.net/jira/software/c/projects/O3/issues/O3-3863?filter=myopenissues&jql=project%20%3D%20%22O3%22%20AND%20assignee%20IN%20%28currentUser%28%29%29%20AND%20statusCategory%20IN%20%28%22To%20Do%22%2C%20%22In%20Progress%22%29%20ORDER%20BY%20created%20DESC)

## Other
The related JIRA ticket also suggests adding E2E tests for the Patient Lists tutorial. However, since the functionality is not working at the moment, I have excluded it from this PR.


[O3-3863]: https://openmrs.atlassian.net/browse/O3-3863?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

harshthakkr committed:
test: Add e2e tests for user onboarding tutorials

harshthakkr committed:
test: Add types

harshthakkr committed:
test: Minor changes

harshthakkr commented on e2e/specs/demo-tutorial.spec.ts:
```diff
@@ -0,0 +1,114 @@
+
+  await test.step('And I click the `Appointments` button', async () => {
+    await page.evaluate(() => {
+      document.querySelector('.react-joyride__overlay')?.setAttribute('style', 'z-index: 1 !important');
```
I had to use this at multiple instances (mostly for a button click) because Playwright was unable to interact with the button as it was covered by a tooltip overlay.

harshthakkr commented:
@jayasanka-sack, @denniskigen Could you please review this PR?

jayasanka-sack commented:
This is awesome @harshthakkr! 😍

@Vijaykv5 @Piumal1999 Could you please review this PR?

jayasanka-sack committed:
Re-trigger E2E tests

jayasanka-sack commented:
@harshthakkr I made a commit to the main branch, could you please take a pull?

harshthakkr commented:
Sure @jayasanka-sack !

harshthakkr commented:
Can you please re-trigger the E2E tests @jayasanka-sack?

Piumal1999 reviewed (dismissed):
Good job! @harshthakkr

I have requested some minor changes.

Btw, the build is failing due to a [playwright issue](https://github.com/microsoft/playwright/issues/33051). Let's bump the Playwright version to `^1.50.1`, which is used in other esm repos.

harshthakkr committed:
test: Requested changes

harshthakkr commented:
@Piumal1999 could you please review it?

harshthakkr commented:
Could you please review it @Piumal1999?
cc: @jayasanka-sack @Vijaykv5

denniskigen committed:
Remove test timeout

Commit:
Align E2E specs and tutorial config with the current app

The app and the platform have moved on since these specs were written, so
this follow-up adapts both the tests and the tutorials they exercise:

- Adapt spec flows to the current tutorial config: Basic Overview is now a
  four-step tutorial, and the patient search bar step no longer
  auto-advances, so the affected specs click Next after typing
- Replace onboarding-test.spec.ts with the per-tutorial
  basic-overview.spec.ts
- Replace the overlay z-index workaround with a clickSpotlightedElement()
  helper that moves the mouse over the spotlighted element first. Joyride
  only releases its overlay (pointer-events: none) after a mousemove inside
  the spotlight, which Playwright's pre-click hit test doesn't trigger
- Search by the generated OpenMRS identifier before clicking a search
  result so parallel test runs can't pick another test's patient
- Fix tutorial targets that no longer exist in the current patient chart:
  the action-menu-patient-chart-items-slot extension slot is gone (the
  chart now uses the workspaces2 action menu), the standalone Start visit
  button was replaced by the Actions > Add visit menu flow, and the
  Siderail step's target no longer renders on the summary page
- Update the help menu button selector; the button no longer has an
  aria-controls attribute
- Add the patient chart, patient banner, vitals, and laboratory apps to
  the E2E environment so the patient chart tutorials can run in CI
- Restore a job-level timeout (30 minutes) on the E2E workflow

All seven specs pass against the dockerized E2E environment.

denniskigen reviewed (approved):
Rebased onto the latest main with fixes for the app drift since March. All seven tutorial specs pass in CI. Thanks @harshthakkr for the great work here!
